lib/client.js

```diff
@@ -227,7 +228,7 @@ class Client extends EventEmitter {
     if (!json)
       throw new Error('Bad response (no body).');
 
-    if (json.error && json.error.length) {
+    if (json.error && res.statusCode >= 400) {
```

Comment:
Okay, this change makes sense as `json.error.length` only would work if the error was a string. However for bcoin it's an object, so this would not throw as mentioned in the PR that added it.